tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -272,23 +272,8 @@ public TracerSettings(IConfigurationSource source)
             }
 
             DbmPropagationLevel dbmPropagationValue;
-            DbmPropagationMode = Enum.TryParse(source?.GetString(ConfigurationKeys.DbmPropagationMode), true, out dbmPropagationValue) ?
-                $"{dbmPropagationValue}".ToLower() : "disabled";
-        }
-
-        /// <summary>
-        /// Gets or sets a value indicating whether the tracer should propagate service data in db queries
-        /// </summary>
-        public enum DbmPropagationLevel
-        {
-            /// <summary>Nothing should be propagated</summary>
-            Disabled,
-
-            /// <summary>Only service tags should be injected</summary>
-            Service,
-
-            /// <summary>Both service details and span traceparent should be injected</summary>
-            Full
+            Enum.TryParse(source?.GetString(ConfigurationKeys.DbmPropagationMode), true, out dbmPropagationValue);
```

Comment:
Nitpicking - This should work as well no? 
Remove the var declaration and replace this line by
Enum.TryParse<DbmPropagationLevel>(source?.GetString(ConfigurationKeys.DbmPropagationMode), true, out var dbmPropagationValue);

Reply:
I see two issues with the current code:
1. I don't see where the result is getting assigned to the `DbmPropagationMode` property. The current line is only assigning to the local variable `dbmPropagationValue`
2. If the enum parsing fails, the behavior is unclear. When `Enum.TryParse` returns false, I think we should directly assign `DbmPropagationMode = DbmPropagationLevel.Disabled`